osgp/platform/osgp-adapter-domain-smartmetering/src/main/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/services/MBusGatewayService.java

```diff
@@ -223,19 +248,32 @@ private void doCoupleMBusDevice(final SmartMeter gatewayDevice,
      * Updates the M-Bus device identified in the input part of the
      * {@code deCoupleMbusResponseDto}.
      *
+     * @param deviceMessageMetadata
      * @param deCoupleMbusDeviceResponseDto
      * @throws FunctionalException
      */
-    public void handleDeCoupleMbusDeviceResponse(final DeCoupleMbusDeviceResponseDto deCoupleMbusDeviceResponseDto)
-            throws FunctionalException {
+    public void handleDeCoupleMbusDeviceResponse(final DeviceMessageMetadata deviceMessageMetadata,
+            final DeCoupleMbusDeviceResponseDto deCoupleMbusDeviceResponseDto) {
+        final Optional<SmartMeter> mbusDeviceFoundOnChannel = this
+                .findByMBusIdentificationNumber(deCoupleMbusDeviceResponseDto.getChannelElementValues());
 
-        final String mbusDeviceIdentification = deCoupleMbusDeviceResponseDto.getmBusDeviceIdentification();
-        final SmartMeter mbusDevice = this.domainHelperService.findSmartMeter(mbusDeviceIdentification);
+        if (!mbusDeviceFoundOnChannel.isPresent()) {
+            return;
```

Comment:
log something? or not needed?

Reply:
maybe cleaner way would be to do mbusDeviceFoundOnChannel.ifPresent(device-> doTheStuff();

Reply:
Can be a valid flow. M-Bus device still referenced from Gateway device's channel but M-Bus device is no longer in use and removed from core database.
Using the ifPresent construction now